js/data.js

```diff
@@ -6,3 +6,11 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+function storeDataInLocalStorage(event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('javascript-local-storage', dataJSON);
+}
+
+var $submit = document.querySelector('form');
+$submit.addEventListener('submit', storeDataInLocalStorage);
```

Comment:
This should be a `beforeunload` event listener, not a `submit` event listener on the `form`. LocalStorage is for saving data for future sessions, which means it only makes sense to update its value right before this session ends. It does not need to be updated every time the form is submitted